homeassistant/components/rabbitair/fan.py

```diff
@@ -0,0 +1,132 @@
+"""Support for Rabbit Air fan entity."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from rabbitair import Mode, Model, Speed
+
+from homeassistant.components.fan import (
+    SUPPORT_PRESET_MODE,
+    SUPPORT_SET_SPEED,
+    FanEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    ordered_list_item_to_percentage,
+    percentage_to_ordered_list_item,
+)
+
+from .const import DOMAIN, KEY_COORDINATOR, KEY_DEVICE
+from .entity import RabbitAirBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+SPEED_LIST = [
+    Speed.Silent,
+    Speed.Low,
+    Speed.Medium,
+    Speed.High,
+    Speed.Turbo,
+]
+
+PRESET_MODE_AUTO = "Auto"
+PRESET_MODE_MANUAL = "Manual"
+PRESET_MODE_POLLEN = "Pollen"
+
+PRESET_MODES = {
+    PRESET_MODE_AUTO: Mode.Auto,
+    PRESET_MODE_MANUAL: Mode.Manual,
+    PRESET_MODE_POLLEN: Mode.Pollen,
+}
+
+
+class RabbitAirFanEntity(RabbitAirBaseEntity, FanEntity):
+    """Fan control functions of the Rabbit Air air purifier."""
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_SET_SPEED
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        await self._set_state(mode=PRESET_MODES[preset_mode])
+
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the speed of the fan, as a percentage."""
+        value = percentage_to_ordered_list_item(SPEED_LIST, percentage)
+        await self._set_state(speed=value)
+
+    async def async_turn_on(
+        self,
+        speed: str | None = None,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn on the fan."""
+        mode_value: Mode | None = None
+        if preset_mode is not None:
+            mode_value = PRESET_MODES[preset_mode]
+        speed_value: Speed | None = None
+        if percentage is not None:
+            speed_value = percentage_to_ordered_list_item(SPEED_LIST, percentage)
+        await self._set_state(power=True, mode=mode_value, speed=speed_value)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the fan off."""
+        await self._set_state(power=False)
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on."""
+        return self.coordinator.data.power is True
```

Comment:
`power` can be `None` according to typing, why not return that?

```suggestion
    @property
    def is_on(self) -> bool | None:
        """Return true if device is on."""
        return self.coordinator.data.power
```